lib/ansible/module_utils/network/iosxr/config/interfaces/interfaces.py

```diff
@@ -100,6 +100,8 @@ def set_state(self, want, have):
         """
         commands = []
         state = self._module.params['state']
+        if state in ('overridden', 'merged', 'replaced') and not want:
+            self._module.fail_json(msg='config is required for state {0}'.format(state))
```

Comment:
msg needs to be more generic in this case
```suggestion
            self._module.fail_json(msg='config parameters are required for state {0}'.format(state))
```